public/login.js

```diff
@@ -0,0 +1,14 @@
+function login(e) {
+  e.target.action = `https://api.core.ac.uk/login_check?continue=${encodeURIComponent(
+    window.location.origin
+  )}`
+}
+
+const urlParams = new URLSearchParams(window.location.search)
```

Comment:
~Do you ming using `var`?~ Sorry, you are using new Web API, that we do not polyfill. Never mind.

Also, could you please what the script into DOM loaded event listener?

Reply:
> Also, could you please what the script into DOM loaded event listener?

I don't get it sorry.

Reply:
To have save save JS execution you should make DOM operations e.g. `Element.addEventListener` after having actual DOM parsed, i.e. after on load event. Check [DOMContentLoaded](https://developer.mozilla.org/en-US/docs/Web/API/Window/DOMContentLoaded_event) event to find more.